provisioning/baremetal_webhook.go

```diff
@@ -31,15 +31,15 @@ func EnsureBaremetalOperatorWebhook(info *ProvisioningInfo) (bool, error) {
 	}
 
 	webhookService := newBaremetalOperatorWebhookService(info.Namespace)
-	_, _, err := resourceapply.ApplyService(info.Client.CoreV1(), info.EventRecorder, webhookService)
+	_, _, err := resourceapply.ApplyService(context.TODO(), info.Client.CoreV1(), info.EventRecorder, webhookService)
 	if err != nil {
 		err = errors.Wrap(err, "unable to create validatingwebhook service")
 		return false, err
 	}
 
 	vw := newBaremetalOperatorWebhook(info.Namespace)
-	expectedGeneration := resourcemerge.ExpectedValidatingWebhooksConfiguration(validatingWebhookConfigurationName, info.ProvConfig.Status.Generations)
-	validatingWebhook, updated, err := resourceapply.ApplyValidatingWebhookConfiguration(info.Client.AdmissionregistrationV1(), info.EventRecorder, vw, expectedGeneration)
+	cache := resourceapply.NewResourceCache()
```

Comment:
This seems like it would work, although we're not getting the caching benefits that we would if it were a persistent global cache.
Incidentally, I suspect you could get the old noCache behaviour back by setting `cache = nil` on the next line.